src/agent/src/namespace.rs

```diff
@@ -46,74 +31,203 @@ pub fn get_current_thread_ns_path(ns_type: &str) -> String {
     )
 }
 
-// setup_persistent_ns creates persistent namespace without switchin to it.
-// Note, pid namespaces cannot be persisted.
-pub fn setup_persistent_ns(logger: Logger, ns_type: &'static str) -> Result<Namespace, String> {
-    if let Err(err) = fs::create_dir_all(PERSISTENT_NS_DIR) {
-        return Err(err.to_string());
+#[derive(Debug)]
+pub struct Namespace {
+    logger: Logger,
+    pub path: String,
+    persistent_ns_dir: String,
+    ns_type: NamespaceType,
+}
+
+impl<'a> Namespace {
```

Comment:
Why we need a lifetime here? I don't understand it

Reply:
It's not needed, I forgot to remove it. It was there as I was fighting the borrow checker.
I'm removing it
Thanks!